parser/js/src/test/scala/hmda/parser/fi/DateGenerators.scala

```diff
@@ -0,0 +1,15 @@
+package hmda.parser.fi
+
+import org.scalacheck.Gen
+import scalajs.js
+
+object DateGenerators {
+  def randomDate: Gen[Int] = {
+    val beginDate = new js.Date("2017-01-01")
+    val endDate = new js.Date("2020-12-31")
+
+    for {
+      randomDate <- Gen.choose(beginDate.getTime(), endDate.getTime())
+    } yield new js.Date(randomDate).getTime().toInt
```

Comment:
Are you sure that is where the problem is?. The `JVM` version produces the same output on the `randomDate` generator, an `Int` (milliseconds) but the tests pass in that case. See [here](https://github.com/jmarin/hmda-platform/blob/scalajs-parser/parser/jvm/src/test/scala/hmda/parser/fi/DateGenerators.scala#L9)